Showcase-iOSTests/About/Interactor/AboutInteractableTests.swift

```diff
@@ -0,0 +1,134 @@
+        aboutInteractor.aboutPresenter = mockAboutPresentable
+        aboutInteractor.dataReference = mockDatabaseRefeceabele
+        interactorUnderTest = aboutInteractor
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        mockError = nil
+    }
+
+    // MARK: Negative tests
+
+    func testThatRetrievingFirbaseLinksReturnsAValidErrorWhenFirebaseLinksRetrieverReturnsAnError() {
+        setUpMockStubs()
+        stub(mockDatabaseRefeceabele) { mock in
+             _ = when(mock.observe(eventType: any(), with: any(), withCancel: any()).then({ _, _, errorCompletion in
+                errorCompletion(self.mockError!)
+            }))
+        }
+        stub(mockAboutPresentable) { mock in
+            _ = when(mock.onRetrieveSocialMediaLinksFailed(with: any()).then({ error in
+                let errorResult = error as NSError
+                XCTAssertEqual(errorResult, self.mockError!)
+            }))
+        }
+        interactorUnderTest?.retrieveSocialMediaLinks()
+        verify(mockAboutPresentable, times(1)).onRetrieveSocialMediaLinksFailed(with: any())
+    }
+
+    func testThatWhenFirebaseLinksRetrieverReturnsANullSnapshotThenOnRetrieveSocialMediaCompletesWithAnEmptySocialMediaLinks() {
+        setUpMockStubs()
+
+        stub(mockDataSnapshot) { (mock) in
+            _ = when(mock.value.get.thenReturn(mockEmptySocialMediaResponse()))
+        }
+
+        stub(mockDatabaseRefeceabele) { mock in
+            _ = when(mock.observe(eventType: any(), with: any(), withCancel: any()).then({ _, successCompletion, _ in
+                successCompletion(self.mockDataSnapshot)
+            }))
+        }
+
+        stub(mockAboutPresentable) { mock in
+            _ = when(mock.onRetrieveSocialMediaLinksComplete(with: any()).then({ (links) in
+                XCTAssertTrue(links.website == nil)
+                XCTAssertTrue(links.instagram == nil)
+                XCTAssertTrue(links.facebook == nil)
+                XCTAssertTrue(links.twitter == nil)
+            }))
+        }
+
+        interactorUnderTest?.retrieveSocialMediaLinks()
+        verify(mockAboutPresentable, times(1)).onRetrieveSocialMediaLinksComplete(with: any())
+    }
+
+    func testThatWhenFirebaseLinksRetrieverReturnsAValidNonEmptySnapShotThenOnRetrievesSocialMediaLinksWithComplete() {
+        setUpMockStubs()
+
+        stub(mockDataSnapshot) { (mock) in
+            _ = when(mock.value.get.thenReturn(mockValidSocialMediaResponse()))
+        }
+
+        stub(mockDatabaseRefeceabele) { mock in
+            _ = when(mock.observe(eventType: any(), with: any(), withCancel: any()).then({ _, successCompletion, _ in
+                successCompletion(self.mockDataSnapshot)
+            }))
+        }
+
+        stub(mockAboutPresentable) { mock in
+            _ = when(mock.onRetrieveSocialMediaLinksComplete(with: any()).then({ (links) in
+                XCTAssertTrue(links.website != nil)
+
```

Comment:
Please assert that you get the write values back that you put in as a mock

Reply:
done